changelog/17.fix.md

```diff
@@ -0,0 +1,3 @@
++ fix computation of cosine decay derivative (we forgot a constant value in the formula)
++ add test `test_cosine_decay_derivative` to ensure that derivative is computed accurately
++ update tests in `test_convergence_intergration` which used the computation of the derivative as test-value
```

Comment:
As the CHANGELOG is user-facing, these can go in trivial (or be deleted). (The logic is that our users should already assume we write tests, so we don't need to tell them everytime)